app/routes/routes.py

```diff
@@ -0,0 +1,266 @@
+        site_session["age"] = form.age.data
+        site_session["name"] = form.name.data
+        site_session["lastname"] = form.lastname.data
+        site_session["gender"] = form.gender.data
+        site_session["profile_description"] = form.profile_description.data
+        site_session["domicile"] = form.domicile.data
+        site_session["education"] = form.education.data
+        site_session["employment_status"] = form.employment_status.data
+
+        return redirect(url_for("user_criteria"))
+
+    elif not form.validate_on_submit() and form.submit_field.data:
+        failure_message = "Przykro mi ale nie udało ci się poprawnie wypełnić formularza, spróbuj ponownie"
+        return render_template(
+            "user-information-form.html", form=form, failure_message=failure_message
+        )
+
+    else:
+        return render_template("user-information-form.html", form=form)
+
+
+@app.route("/user_criteria", methods=["GET", "POST"])
+def user_criteria():
+    form = UserCriteria()
+    if form.validate_on_submit():
+        site_session["type_of_robot"] = form.type_of_robot.data
+        site_session["distance"] = form.distance.data
+        site_session["employment_status_criteria"] = form.employment_status.data
+
+        return redirect(url_for("get_main_image"))
+    return render_template("user-criteria-form.html", form=form)
+
+
+@app.route("/get_main_image", methods=["GET", "POST"])
+def get_main_image():
+    form = UploadForm()
+    if form.validate_on_submit():
+        username = site_session.get("username")
+        name = site_session.get("name")
+        lastname = site_session.get("lastname")
+        email = site_session.get("email")
+
+        password = site_session.get("password")
+        print(password)
+        #user_password_decrypted = cipher_suite.decrypt(password)
+        #print(user_password_decrypted)
+        #user_password_decoded = user_password_decrypted.decode("utf-8")
+        #print(user_password_decoded)
+
+        age = site_session.get("age")
+        gender = site_session.get("gender")
+        profile_description = site_session.get("profile_description")
+        domicile = site_session.get("domicile")
+        education = site_session.get("education")
+        employment_status = site_session.get("employment_status")
+        type_of_robot = site_session.get("type_of_robot")
+        distance = site_session.get("distance")
+        employment_status_criteria = site_session.get("employment_status_criteria")
+
+        image_name = get_image(form.photo.data, name)
+
+        insert_user_and_user_profile(
+            username,
+            name,
+            lastname,
+            password,
+            email,
+            image_name,
+            age,
+            gender,
+            profile_description,
+            domicile,
+            education,
+            employment_status,
+            type_of_robot,
+            distance,
+            employment_status_criteria,
+        )

```

Comment:
you can use dict unpacking here

I would validate arguments with a form to be sure that they were given